testsuite/ext-tools/json_creation_script.py

```diff
@@ -0,0 +1,168 @@
+
+    args = parser.parse_args()
+    return args
+
+
+def read_requests():
+    # Find open requests
+    result = object
+    try:
+        # TODO Find a better way to query the open requests, this is fragile because it depends on external utils
+        #  being there.
+        result = subprocess.run(["osc --apiurl https://api.suse.de qam open -G qam-manager"], shell=True,
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    except subprocess.CalledProcessError:
+        print("The ibs command failed for some reason")
+    output = result.stdout.decode('utf-8')
+    lines = output.splitlines()
+    # Create empty list to add the rrids from the output
+    rrids = []
+    for line in lines:
+        if "ReviewRequest" in line:
+            line1 = line.rstrip()
+            line2 = line1.split(sep=":")
+            rrid = line2[3]
+            rrids.append(rrid)
+    return rrids
+
+
+def find_valid_repos(rrids, version):
+    if version == '42':
+        dict_version = nodesdict42
+    elif version == '43':
+        dict_version = nodesdict43
+    else:
+        print("You have not given one of the correct options, run the script with -h to see the correct ones")
+        sys.exit(1)
+
+    finaldict = {}
+    for node, suffixraw in dict_version.items():
+        for rrid in rrids:
+            if isinstance(suffixraw, str):
+                suffix = suffixraw
+                repo = create_url(rrid, suffix)
+                if repo is not None:
+                    # For multiple rrids that affect same node, eg server has multiple requests:
+                    if node in finaldict:
+                        finaldict[node][rrid] = repo
+                    else:
+                        finaldict[node] = {rrid: repo}
+            elif isinstance(suffixraw, list):
+                for suffix in suffixraw:
+                    repo = create_url(rrid, suffix)
+                    if repo is not None:
+                        # For multiple rrids that affect same node, eg server has multiple requests:
+                        if node in finaldict:
+                            finaldict[node][rrid] = repo
+                        else:
+                            finaldict[node] = {rrid: repo}
+
+    # Format into json and print
+    # Check that it's not empty and save to file
+    if finaldict:
+        with open('custom_repositories.json', 'w', encoding='utf-8') as f:
+            json.dump(finaldict, f, indent=2)
+    else:
+        print("Dictionary is emtpy, something went wrong")
+        sys.exit(1)
+
+
+def create_url(rrid, suffix):
+    link = ["http://download.suse.de/ibs/SUSE:/Maintenance:/", str(rrid)]
+
+    if link[:-1] == str(rrid):
+        link.append(suffix)
+        url = ''.join(link)
+    else:
+        link = ''.join(link)
+        url = str(link) + suffix
+    re = requests.get(url)
+    if re.ok:
+        return url
+
+
+def main():
+    args = parse_args()
+    rrids = read_requests()
```

Comment:
I suggest to provide a mechanism to pass a list of rrids manually.
And only optionally, obtain them using osc.

The reason is that we might not want to include all the MIs available for a certain version, but only a concrete bunch of submissions packaged as part of a Maintenance Update. In that way, we can control if we want to test extra MIs not part of the MU (for instance those packages that we need to approve but we don't have control of the day it is released).

Reply:
I agree with Oscar's view. I'd not include all the pending SLE updates into the JSON.

MIs from the JIRA ticket have one release date, MIs from SLE have another release date.

We should not block our SUSE Manager releases because those packages from SLE, and we can't strictly block those packages for SLE until we release SUSE Manager (unless we find an issue).

We can find issues testing on what's released, say 4.2.7. That's what we are doing now AFAIK.

Then that SLE update should work on what's released  (4.2.7) and on what will be released (another MU that we test at that given time, say 4.2.8), because it was already working.

Reply:
That requires a big refactor, I have to schedule this after my vacation unfortunately. What I recommend now is to checkout the PR when you need to run the script and run it after clearing the queue of irrelevant updates, and maybe assign yourself to the ones that need testing to bypass having them in the open requests. That way it will run only for the SUMA update we want it to.

Reply:
This is still pending, I will get to it soon, for now only the workaround of clearing the queue beforehand exists. I talked with Marina and there is a way that maintenance can differentiate between SUMA requests and others but that's for a later step.

Reply:
Ok, but we can't accept this PR until we can't manually pass a list of MIs.
That's not a big refactor, it's very easy in fact, you can found the example in my crawler script.
Or if you prefer I can contribute in this PR to add support for this.

Reply:
Thanks for fixing this, now we can pass a comma separated list of MUs 💯